### PR TITLE
feat(internal): cache extract list

### PR DESCRIPTION
**Pull request by rarkins**

### CODE REVIEW

File: lib/workers/repository/extract/cache.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import crypto from 'crypto';
+import { RenovateConfig } from '../../../config/common';
+import { logger } from '../../../logger';
+import { PackageFile } from '../../../manager/common';
+
+function getCacheNamespaceKey(
+  config: RenovateConfig
+): { cacheNamespace: string; cacheKey: string } {
+  // Cache extract results per-base branch
+  const { platform, repository, baseBranch } = config;
```
Comment (viceice) [inline lib/workers/repository/extract/cache.ts:10]: Can we do the destructuring in function decalration?

**Review by viceice — COMMENTED** [review]
Needs test fixes too. Generally LGTM, but can we do object destruction in function declaration instead of separate const ?

**Comment by rarkins** [comment]
- [ ] Need to check that this doesn't interfere with Gradle's extraction and lookup

**Comment by viceice** [comment]
@rarkins Isn't this obsolete by `repositoryCache` now? 🤔

**Pull request closed without merging**